src/main/java/net/spy/memcached/CacheMonitor.java

```diff
@@ -51,18 +54,29 @@ public class CacheMonitor extends SpyObject implements Watcher,
    * @param serviceCode     service code (or cloud name) to identify each cloud
    * @param listener        Callback listener
    */
-  public CacheMonitor(ZooKeeper zk, String cacheListZPath, String serviceCode,
-                      CacheMonitorListener listener) {
+  public CacheMonitor(ZooKeeper zk, String cacheListZPath,
+                      String serviceCode, boolean startup,
+                      CacheMonitorListener listener) throws InterruptedException {
     this.zk = zk;
     this.cacheListZPath = cacheListZPath;
     this.serviceCode = serviceCode;
     this.listener = listener;
 
+    if (startup) {
+      this.clientInitLatch = new CountDownLatch(1);
+    } else {
+      this.clientInitLatch = new CountDownLatch(0);
+    }
+
     getLogger().info("Initializing the CacheMonitor.");
 
     // Get the cache list from the Arcus admin asynchronously.
     // Returning list would be processed in processResult().
     asyncGetCacheList();
+
+    if (startup) {
+      this.clientInitLatch.await();
+    }
```

Comment:
`startup == false`인 경우 Latch가 0인 상태인데
await이 단순 비교에 비해 cost가 큰 동작이라서 if문을 사용하는 것인가요?

Reply:
`if (startup)` 조건을 사용하는 상태입니다.
다른 걸 얘기하려는 것인가요?

Reply:
```suggestion
      this.clientInitLatch.await();
```
위와 같이 if문 없이 await만 사용하는 것과 차이가 있나요?
- `startup == true` => Latch가 0이 될 때까지 대기
- `startup == false` => Latch가 이미 `0`. 바로 pass

Reply:
아. 무슨 말인지 이해했어요.
cost 차이는 없습니다.
 startup 시에만 await하는 것을 명백히 하기 위해, if 조건을 두는 것이 좋을 것 같습니다.